### ISSUE TITLE
[Feature Request] Tournaments near me

### ISSUE THREAD

**Issue by gereons** [madarasz/always-be-running#42]
Instead of (optionally) restricting the view for upcoming tournaments to a user's country, it would be great to have the option to say e.g. "show me everything in a 200km radius". 

Use case: I'm in Cologne, fairly close to the Netherlands and Belgium border, so tournaments in Eindhoven and Liége are much closer and attractive to me than those in Berlin or Munich.

**Comment by GuvBubbs** [comment]
I like this. If there isn't enough in your home country then it fills with the next closest.

**Comment by madarasz** [comment]
Hmm... Yeah, you can do this geolocation thing in the browser.  Though I like to keep it simple. What should be the default distance? I want to spare the UI from another field. 200km sounds reasonable in Europe. I'm not so sure about the rest of the world, especially US.

Donald: "closest country" is trivial from New Zealand, anywhere else it's not :D I could try "closest" but you would still not be happy if it found a tournament next door but a year later...

**Comment by gereons** [comment]
I don't think this will work. My IP-based geolocation is somewhere in the outskirts of Cologne, and as there will always be an upcoming tournament in downtown Cologne, this would reduce the distance to 20 or 30km. Doubling that wouldn't even show me events in Aachen or Dortmund which are two of the closest cities that run tournaments.

I would actually prefer to have two user profile fields, home town and distance for this.

**Comment by madarasz** [comment]
I wrote in point "2" to use minimum distance of 150km, which is doubled. I guess you misunderstood me. So the search would be 300km minimum and twice the distance to the closest tournament.

**Comment by gereons** [comment]
Yeah, sorry, I totally misunderstood the "use that" to mean the "closest distance". :)

**Comment by ANR-PeterPan** [comment]
perhaps we can get a "europe" Country and/or US states too?

**Comment by madarasz** [comment]
@gereons @GuvBubbs @ANR-PeterPan: I implemented a "Zoom to me" button on the Upcoming page. Please check it out if it works.

Limitations:
- Doesn't for for Safari browsers
- Current tournament filtering is not affected by the button. This is only a zooming function.

**Comment by gereons** [comment]
Works ok for me, but I would change two things: 

a) make it zoom in 1 or 2 levels deeper. At the current zoom level, I still see everything from Cardiff to Vienna.

b) automatically remove the country selection, because for people near borders (i.e., everyone in continental Europe) stopping at borders kinda defeats the purpose :)

**Comment by 5N00P1** [comment]
Agree with @gereons as I have to manually select the countries close to me, which is Poland & Czechia.

**Comment by madarasz** [comment]
@gereons: thx for the feedback. "A" is a bug, "B" will be implemented later with the improved filtering feature (the browser will download all tournaments once and do the filtering in the frontend, currently filtering is done by the backend)
